chatlearn/models/base_module.py

```diff
@@ -728,7 +730,8 @@ def allgather_expert_parameter(self, group_name, pipe_stage=0):
         for name, param in self._parameters_to_sync[pipe_stage]:
             param, state = self._synchronizer.regroup_experts_from_all_tp_ranks(name, param, self.tensor_parallel_rank(), group_name)
             if state:
-                self._expert_sync_buffer[name].append(param)
+                self._expert_sync_buffer.pop(name, "Not Found.")
```

Comment:
should we throw an exception if name is not found

Reply:
`pop` here aims to release some weight in _expert_sync_buffer, if not found, pass is ok